OOP/assignment3/RailwayReservationSystem.java

```diff
@@ -0,0 +1,321 @@
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Scanner;
+
+public class RailwayReservationSystem {
+	public static ArrayList<Train> trainList = new ArrayList<Train>();
+
+	private static void prepareTrain() {
+
+		final String COMMA_DELIMITER = ",";
+		final String url = "D://Train/TrainDetails.csv";
+
+		BufferedReader fileReader = null;
+		try {
+
+			String line = "";
+			fileReader = new BufferedReader(new FileReader(url));// file reading
+			fileReader.readLine();
+
+			while ((line = fileReader.readLine()) != null) {
+
+				Train train = new Train();
+				String[] tokens = line.split(COMMA_DELIMITER);// separator
+
+				if (tokens.length > 0) {
+					train.setName(tokens[0]);
+					train.setFromto(tokens[1]);
+					train.setDuration(Integer.parseInt(tokens[2]));
+					train.setAvail(Integer.parseInt(tokens[3]));
+
+					if (tokens[4].equals("PASSENGER"))
+						train.type = Train.Type.PASSENGER;
+					else
+						train.type = Train.Type.GOODS;
+				}
+				trainList.add(train);
+
+			}
+		} catch (FileNotFoundException e) {
+			System.out.println("File not found");
+		} catch (IOException e) {
+			System.out.println("error in input");
+		} finally {
+
+			try {
+				fileReader.close();
+			} catch (IOException e) {
+				System.out.println("error in file input");
+
+			}
+
+		}
+
+		Collections.sort(trainList, Train.trainDuration);
+	}
+
+	private static void showChart(Train.Type type, String fromto) {
+		for (Train train : trainList) {
+			if (train.type.equals(type) && train.getFromto().equals(fromto)) {
+
+				System.out.println(train.getName() + "    "
+						+ train.getDuration() + " hrs" + "    "
+						+ train.getAvail());
+
+			}
+
+		}
+
+	}
+
+	private static void showChart(Train.Type type) {
+
+		for (Train train : trainList) {
+			if (train.type.equals(type)) {
+
+				System.out.println(train.getName() + "    "
+						+ train.getDuration() + " hrs" + "    "
+						+ train.getAvail() + "    " + train.getFromto());
+
+			}
+
+		}
+
+	}
+
+	private static void printTicket(String name, String trainName,
+			int noOfSeat, int amount) {
+
+		System.out.println("Name  : " + name);
+		System.out.println("train name  : " + trainName);
+		System.out.println("no of seat  : " + noOfSeat);
+		System.out.println("Amount  : " + amount);
+	}
+
+	private static int getAvailable(String trainName) {
+		for (Train train : trainList) {
+			if (train.getName().equals(trainName))
+				return train.getAvail();
+
+		}
+
+		return -1;
+
+	}
+
+	private static void InputProcess() {
+		while (true) {
+			Scanner scanner = new Scanner(System.in);
+			System.out
+					.println("welcome to Railways ticket booking center Enter your name");
+			String name = scanner.next();
+			System.out.println("enter train type\n1-Passenger\n2-goods");
+			int input = scanner.nextInt();
+			Train.Type type;
+			if (input == 1)
```

Comment:
Such kind of simple if-else statements (involving assignment to one variable) can implemented using conditional statements.